src/posts/xwrf/index.md

```diff
@@ -0,0 +1,162 @@
+3. generating projection coordinates and providing a [`pyproj.CRS`](https://pyproj4.github.io/pyproj/dev/api/crs/crs.html) object
+
+In this post, we will show how `xWRF` works together with other utilities in order to post-process a large `WRF` output dataset.
+
++++
+
+## Installing `xWRF`
+
+Before we start using `xWRF`, we need to install it. We can do this by following these steps!
+
+1. Install this in your python environment using pip (`pip install xwrf`)
+2. Open up a notebook and use the imports shown below!
+
+```{code-cell} ipython3
+import holoviews as hv
+import hvplot
+import hvplot.xarray
+import xarray as xr
+import xwrf
+from distributed import Client
+from dask_jobqueue import SLURMCluster
+import pint_xarray
+import numpy as np
+
+hv.extension('bokeh')
+```
+
+## Spin up a Cluster
+
+```{code-cell} ipython3
+cluster = SLURMCluster(name='dask-cluster',
+                      cores=128,
+                      memory='125G',
+                      processes=11,
+                      interface='ib0',
+                      queue='compute',
+                      walltime='8:00:00',
+                      asynchronous=0)
+n_workers = 2
+cluster.scale(n_workers)
+client = Client(cluster)
+client.wait_for_workers(n_workers=n_workers)
+client
+```
+
+## Load the WRF output
+
+```{code-cell} ipython3
+import intake
+cat = intake.open_catalog("https://raw.githubusercontent.com/xarray-contrib/xwrf-data/main/catalogs/catalog.yml")
+ssp5_ds = cat["xwrf-sample-ssp585"].to_dask()
+ssp5_ds
+```
+
+As one can see, this output is not very helpful. Heaps of necessary information like WRF model grid coordinates or coordinate index assignments are missing. But we can use `xWRF` to very efficiently fix this.
+
+```{code-cell} ipython3
+%time
+ssp5_ds = ssp5_ds.xwrf.postprocess()
+ssp5_ds
+```
+
+As you can see, the `xWRF` postprocessing is done in no time at all thanks to it leveraging the `dask`'s integration and it's delayed computation feature. This is despite the uncompressed dataset being close to 60GB in size.
+
+```{code-cell} ipython3
+ssp5_ds.nbytes*1E-9
+```
+
+Next to some cleanup of the coordinates, the `xWRF` postprocessing includes the calculation of some basic diagnostics not included in WRF output, namely `air_pressure`, `air_potential_temperature`, `geopotential` and `geopotential_height`.
+
+Using `dask`, the computations for determining these diagnostics are delayed. They will be executed upon using `.compute()`, `.persist()` or `.values` on the variable itself or a variable depending on it.
+
+```{code-cell} ipython3
+ssp5_ds.air_pressure
+```
+
+## Example analysis
+
++++
+
+Let's just imagine that we would want to investigate e.g. how climate change might affect the jet stream in different SSP scenarios. One comparison we might want to make could be between the SSP 585 and the SSP 245 simulations.
+
+So first, we load the SSP 245 dataset, too.
```

Comment:
An explanation of what these different SSP values mean would be helpful - even just using one of the figures from the IPCC report, and link to more info.

Reply:
Added some more info